buildenv/jenkins/jobs/pipelines/Pipeline-Build-Test-All

```diff
@@ -139,19 +168,124 @@ try {
                     variableFile = load 'buildenv/jenkins/common/variables-functions'
                     buildFile = load 'buildenv/jenkins/common/pipeline-functions'
 
+                    // Determine if build is a PullRequest
+                    if (ghprbPullId) {
+                        // Parse Github trigger comment
+                        // Jenkins test sanity <platform>*
+                        // Jenkins test extended <platform>*
+                        // Jenkins compile <platform>*
+                        // Jenkins test <comma_separated_test_target> <platform> *
+                        //params.TESTS_TARGETS = params.ghprbCommentBody.substring(?)
+                        /*
+                        * Comment indexes - '+' is an offset of 0/1 if running tests or not
+                        * 0 - Jenkins
+                        * 1 - test / compile
+                        * 1+ - test target
+                        * 2+ - platforms
+                        * 3+ - jdk versions
+                        */
+                        def PARSED_COMMENT = params.ghprbCommentBody.toLowerCase().tokenize(' ')
+                        if (PARSED_COMMENT[0] != 'adam') {
+                            error("Pull Request trigger comment needs to begin with 'Jenkins'")
+                        }
+
+                        // Verify min comment length
+                        if (PARSED_COMMENT[1] == 'compile') {
+                            if (PARSED_COMMENT.size() < 4) {
```

Comment:
The inner `if (PARSED_COMMENT.size() < *) {...}` can moved into the `switch` statement below.